tests/src/test/java/org/eclipse/hono/tests/registry/TenantApiTests.java

```diff
@@ -65,41 +66,47 @@
+        final JsonObject httpAdapterExtensions = new JsonObject()
+                .put("deployment", new JsonObject()
+                        .put("maxInstances", 4));
+
+        final JsonObject dataVolume = new JsonObject()
                         .put("max-bytes", 2147483648L)
                         .put("period-in-days", 30)
-                        .put("effective-since", "2019-04-27T12:00:00+00:00"));
-        final JsonArray adapterConfig = new JsonArray()
-                .add(new JsonObject()
-                        .put(TenantConstants.FIELD_ADAPTERS_TYPE, Constants.PROTOCOL_ADAPTER_TYPE_MQTT)
-                        .put(TenantConstants.FIELD_ENABLED, true)
-                        .put(TenantConstants.FIELD_ADAPTERS_DEVICE_AUTHENTICATION_REQUIRED, true))
-                .add(new JsonObject()
-                        .put(TenantConstants.FIELD_ADAPTERS_TYPE, Constants.PROTOCOL_ADAPTER_TYPE_HTTP)
-                        .put(TenantConstants.FIELD_ENABLED, true)
-                        .put(TenantConstants.FIELD_ADAPTERS_DEVICE_AUTHENTICATION_REQUIRED, true)
-                        .put("deployment", new JsonObject()
-                                .put("maxInstances", 4)));
+                        .put("effective-since", "2019-04-27T12:00:00+00:00");
+
+        final ResourceLimits resourceLimits = new ResourceLimits();
+        resourceLimits.setMaxConnections(100000);
+        if (resourceLimits.getExtensions() == null) {
+            resourceLimits.setExtensions(new HashMap<>());
+        }
+
+        resourceLimits.getExtensions().put("data-volume", dataVolume.getMap());
+
         final String tenantId = getHelper().getRandomTenantId();
-        final TenantObject tenant = TenantObject.from(tenantId, true)
-                .setResourceLimits(resourceLimits)
-                .setDefaults(defaults)
-                .setAdapterConfigurations(adapterConfig)
-                .setProperty("customer", "ACME Inc.");
+        final Tenant tenant = new Tenant();
+        tenant.setEnabled(true);
+        tenant.setLimits(resourceLimits);
+        tenant.setDefaults(defaults.getMap());
+        if (tenant.getExtensions() == null) {
+            tenant.setExtensions(new HashMap<>());
+        }
+        tenant.getExtensions().put("customer", "ACME Inc.");
+        Tenants.setAdapterEnabled(tenant, Constants.PROTOCOL_ADAPTER_TYPE_MQTT, true);
+        Tenants.setAdapterEnabled(tenant, Constants.PROTOCOL_ADAPTER_TYPE_HTTP, true, true, httpAdapterExtensions.getMap());
 
         getHelper().registry
-        .addTenant(JsonObject.mapFrom(tenant))
+        .addTenant(tenantId, tenant)
         .compose(ok -> getAdminClient().get(tenantId))
         .setHandler(ctx.succeeding(tenantObject -> {
             ctx.verify(() -> {
-                assertThat(tenantObject.isEnabled()).isTrue();
+                final Tenant convertedTenantObject = Tenants.convertTenantObject(tenantObject);
```

Comment:
why do you want to convert the returned object here? The object has been retrieved using the `TenantClient` which is used by the protocol adapters. IMHO it would make more sense to verify, that the data returned by the registry contains all data that has been put into it using the Management API when we set up the fixture for this test.

Reply:
that make sense 👍 removing the conversion will answer the following question

> is there a particular reason why you override hash and equals?